samples/ehr-service/lambda/auth/index.js

```diff
@@ -41,25 +58,40 @@ function authorizeRequest(userToken, serviceToken, path, method, methodArn, call
     console.log('auth service response time - auth public endpoint: ' + latency + 'ms')
     let decoded = ''
     if (resp.status == 200) {
-      decoded = jwt.decode(userToken)
-      callback(null, generatePolicy(decoded.sub, 'Allow', methodArn, decoded.accountId, latency))
+      if (!userToken) {
+        // The path accessed without token found in exclusion list
+        // hence ARC returned 200.
+        callback(null, generatePolicy("USER", "Allow", methodArn, "", latency));
+      } else {
+        // A valid access token was found
+        let decoded = jwt.decode(userToken)
+        callback(null, generatePolicy(decoded.sub, "Allow", methodArn, decoded.accountId, latency));
+      }
     } else {
+      // No token was found for a path which is not in exclusion list for the service
+      // Need to redirect for login
       if(!userToken) {
+        let state = "https://" + e.requestContext.domainName + e.requestContext.path
```

Comment:
I think authorizeRequest() is doing a lot of things. We should separate out the actual authorize (200 vs. 302 vs. 401) logic in the handler and writer helper methods for the workflow. This will make the code more readable and developers can follow along better.